src/rfid.cpp

```diff
@@ -0,0 +1,460 @@
+/*
+ * Author: Oliver Norin
+ *         Matthias Akstaller
+ */
+
+#if defined(ENABLE_DEBUG) && !defined(ENABLE_DEBUG_NFCREADER)
+#undef ENABLE_DEBUG
+#endif
+
+#include "rfid.h"
+
+#include "debug.h"
+#include "mqtt.h"
+#include "lcd.h"
+#include "app_config.h"
+#include "input.h"
+#include "openevse.h"
+
+#include <Wire.h>
+
+#ifndef I2C_SDA
+#define I2C_SDA 21
+#endif
+
+#ifndef I2C_SCL
+#define I2C_SCL 22
+#endif
+
+/*
+ * Documentation of the I2C messages:
+ * NXP PN532 manual, https://www.nxp.com/docs/en/user-guide/141520.pdf
+ * (accessed: 2022-01-14)
+ */
+
+#define PN532_I2C_ADDRESS  (0x48 >> 1)
+#define PN532_RDY 0x01
+#define PN532_FRAME_HEADER_LEN 4
+#define PN532_FRAME_IDENTIFIER_LEN 1
+#define PN532_CMD_SAMCONFIGURATION_RESPONSE 0x15
+#define PN532_CMD_INAUTOPOLL_RESPONSE 0x61
+
+#define I2C_READ_BUFFSIZE 35 //might increase later
+
+RfidTask::RfidTask() :
+  MicroTasks::Task()
+{
+}
+
+void RfidTask::begin(EvseManager &evse){
+    _evse = &evse;
+    Wire.begin(I2C_SDA, I2C_SCL);
```

Comment:
Probably this is an idea for a new feature? It would be more elegant to initialize `Wire` in `main.cpp` and just pass a pointer to the RFID class. However, the indirection here has no real benifit if only the RFID class uses it. But if you want I can do it, it only takes a few minutes.

Anyway, the `TwoWire` class handles multiple calls of `begin(...)` without problem.

Reply:
That shouldn't be a problem then